src/main/java/io/mosip/mimoto/controller/IssuersController.java

```diff
@@ -63,24 +80,40 @@ public ResponseEntity<CredentialIssuerWellKnownResponse> getIssuerWellknown(@Pat
     @GetMapping(value = "/{issuer-id}", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<ResponseWrapper<IssuerDTO>> getIssuerConfig(@PathVariable("issuer-id") String issuerId) {
         ResponseWrapper<IssuerDTO> responseWrapper = new ResponseWrapper<>();
-        IssuerDTO issuerConfig;
+        IssuerDTO issuerDTO;
         try {
-            issuerConfig = issuersService.getIssuerConfig(issuerId);
+            issuerDTO = issuersService.getIssuerDetails(issuerId);
+            responseWrapper.setResponse(issuerDTO);
+            return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+        } catch (InvalidIssuerIdException exception) {
+            log.error("invalid issuer id passed - ", issuerId);
+            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
         } catch (Exception exception) {
             log.error("Exception occurred while fetching issuers ", exception);
-            String[] errorObj = Utilities.handleExceptionWithErrorCode(exception);
+            String[] errorObj = Utilities.handleExceptionWithErrorCode(exception, INVALID_ISSUER_ID_EXCEPTION.getCode());
             List<ErrorDTO> errors = Utilities.getErrors(errorObj[0], errorObj[1]);
             responseWrapper.setResponse(null);
             responseWrapper.setErrors(errors);
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(responseWrapper);
         }
-        responseWrapper.setResponse(issuerConfig);
-        if (issuerConfig == null) {
-            log.error("invalid issuer id passed - {}", issuerId);
-            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
-        }
+    }
 
-        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    @Operation(summary = SwaggerLiteralConstants.ISSUERS_GET_ISSUER_CONFIGURATION_SUMMARY, description = SwaggerLiteralConstants.ISSUERS_GET_ISSUER_CONFIGURATION_DESCRIPTION)
+    @GetMapping(value = "/{issuer-id}/configuration", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Critical : who is the consumer of this endpoint ? For inji web ui, do we need this full set ? Can we not live with only selected details ? That too can we not give it will the above getIssuerConfig endpoint ?